dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/OrgUnitValueTypeSupplier.java

```diff
@@ -87,4 +99,15 @@ private void collectResourceIds( List<String> orgUnitAttributes, List<String> or
             }
         } );
     }
+
+    private void collectResourceIds( List<String> orgUnitDataElements, List<String> orgUnitIds,
+        Set<DataValue> dataElements )
+    {
+        dataElements.forEach( de -> {
+            if ( orgUnitDataElements.contains( de.getDataElement() ) && !StringUtils.isEmpty( de.getValue() ) )
```

Comment:
Why don't we already filter by type and query only for the org unit ids?

Reply:
orgUnitDataElements is already filtered one. it contains all dataElements of ou value type. ~~dataElements~~ > dataValues.